google/cloud/aiplatform/datasets/dataset.py

```diff
@@ -482,21 +457,22 @@ def import_data(
                 labels specified inside index file refenced by
                 [import_schema_uri][google.cloud.aiplatform.v1beta1.ImportDataConfig.import_schema_uri],
                 e.g. jsonl file.
-            sync (bool):
+            sync: (bool) = True
                 Whether to execute this method synchronously. If False, this method
                 will be executed in concurrent Future and any downstream object will
                 be immediately returned and synced when the Future has completed.
+
         Returns:
             dataset (Dataset):
                 Instantiated representation of the managed dataset resource.
         """
-
-        import_lro = self._import_from_gcs(
-            source=gcs_source,
-            import_schema_uri=import_schema_uri,
-            data_items_labels=data_items_labels,
+        # Validate metadata_schema_uri and import_schema_uri for Dataset class
+        self._validate_import_schema_uri(import_schema_uri)
+        datasource = datasources.NonTabularDatasourceImportable(
```

Comment:
This is one exception to the dependency-inversion principle I talked about. A rare trade-off to maintain current user experience.